pkg/autoscaler/statforwarder/forwarder.go

```diff
@@ -48,11 +48,22 @@ const (
 	autoscalerPortName = "http"
 	retryTimeout       = 3 * time.Second
 	retryInterval      = 100 * time.Millisecond
+
+	// Retry as most 15 seconds to process a stat. NOTE: Retrying could
```

Comment:
sorry, nit:

```suggestion
	// Retry at most 15 seconds to process a stat. NOTE: Retrying could
```

Reply:
One thing I worry about is I think we bucket the events by time we record them, not by time they "occurred". With a fast path from QP to autoscaler that's not a big deal, but as that lag becomes bigger it becomes a bit more of an issue. The particular thing that could go wrong here is if we have a blip that causes a whole set of events to all land at time N after a delay: they'll all be bucketed together and averaged, which I think is not what we want. Having said that this could happen in the current code with a network burp, in theory, so maybe it's fine.

Reply:
I agree so I added the NOTE part. I could see this happens in two cases: 1) The lease is waiting to be acquired. This could happen when upgrading a cluster and all old pods are shutdown and new pods are set up. The new pods need to wait `LeaseDuration`, which is 15 seconds by default. 2) With mesh, the owner pod crashes and another pod takes over. The IP for the Endpoints changes but it takes several seconds to complete the network programming.

We can revisit this if we see this is really a problem.